src/dataset.jl

```diff
@@ -172,13 +172,111 @@ end
     write(dataset::AbstractDataset, filename::AbstractString; kwargs...)
 
 Writes the dataset to the designated filename.
+
+### Parameters
+* `dataset`: The dataset to write
+* `filename`: The filename, UTF-8 encoded.
+
+### Keyword Arguments
+* `driver`: The driver to use, you have to manually select the right driver via `getdriver(drivername)` matching the file extension you wish. 
+Otherwise the driver of the source dataset will be used.
+* `options` (Vector{String}): A vector of strings containing KEY=VALUE pairs for driver-specific creation options.
+* `layer_options` (Vector{String}): Driver specific options for layer creation when writing vector datasets.
+* `chunksize` (Integer): Number of features to write in one database transaction. Neglected when `use_gdal_copy` is true.
+* `use_gdal_copy` (Bool): Set this to `true` if you encounter errors when writing. Set to `false` for higher speed.
+* `strict` (Bool): Set this to `true` if the written dataset should be a 1:1 copy of the source data, default is `false`,
+which allows the driver to adapt if necessary.
+    
+### Returns
+`nothing`
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
-    destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    if nraster(dataset) > 0 && nlayer(dataset) > 0
+        drivername = :driver ∈ keys(kwargs) ? shortname(kwargs[:driver]) : shortname(getdriver(dataset))
+        error("Writing datasets with raster and vector data is not supported when using driver $drivername. 
+        Please file an issue at https://github.com/yeesian/ArchGDAL.jl/issues 
+        including following dataset information: \n\n$dataset")
+    elseif nraster(dataset) > 0
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif nlayer(dataset) > 0
+        writelayers(dataset, filename; kwargs...)
+    end
+    return nothing
+end
+
+"""
+    writelayers(ds, filename; driver, options=[""], layer_options=[""], chunksize=20000)
+
+Writes the vector dataset to the designated filename. The options are passed to the newly created dataset and
+have to be given as a list of strings in KEY=VALUE format. The chunksize controls the number of features written 
+in each database transaction, e.g. for SQLite. This function can also be used to copy datasets on disk.
+
+Currently working drivers: FlatGeobuf, GeoJSON, GeoJSONSeq, GML, GPKG, JML, KML, MapML, ESRI Shapefile, SQLite
+
+### Parameters
+*`dataset`  The source dataset
+*`filename` The file name to write to
+
+### Keyword arguments
+`driver`           The driver to use, you have to manually select the right driver for the file extension you wish
+`options`   A vector of strings containing KEY=VALUE pairs for driver-specific creation options
```

Comment:
Should this be named `dataset_options` (to distinguish it from `layer_options`)?

Reply:
Then you would still need the `options` kwarg for the raster writer or fiddle with the kwargs to rename `dataset_options` to `options`.